### PR TITLE
Fixes List only API products purchased by a team

### LINKED ISSUE #336: List only API products purchased by a team
**Is your feature request related to a problem? Please describe.**
Customer noticed some feature not implemented yet in Drupal Module, regarding teams in monetization:

In a team project you can see API products (rate plans) which are not purchased by the team.

**Describe the solution you'd like**
The customer would like to see only the purchased rate plan in the project.

### PR DESCRIPTION
**Pull request by kedarkhaire**
Fixes #336

### CODE REVIEW

File: apigee_m10n.module [hunk 1]
```diff
@@ -40,6 +40,20 @@ function apigee_m10n_api_product_access(EntityInterface $entity, $operation, Acc
   return AccessResult::neutral();
 }
 
+/**
+ * Implements hook_apigee_edge_teams_team_api_product_access().
+ */
+function apigee_m10n_teams_apigee_edge_teams_team_api_product_access(ApiProductInterface $api_product, string $operation, TeamInterface $team, AccountInterface $account) {
```
Comment (divya-intelli) [inline apigee_m10n.module]: Hook name is not correct, you have added in m10n.module. I think this needs to be added in m10n teams module.

File: src/Monetization.php [hunk 2]
```diff
@@ -237,6 +239,48 @@ public function apiProductAssignmentAccess(EntityInterface $entity, AccountInter
     }
   }
 
+  /**
+   * {@inheritdoc}
+   */
+  public function apiProductTeamAssignmentAccess(ApiProductInterface $api_product, TeamInterface $team, AccountInterface $account): AccessResultInterface {
```
Comment (divya-intelli) [inline src/Monetization.php]: These changes needs to be added in m10n teams module.

File: apigee_m10n.module (hunk 1 above)
Comment (kedarkhaire) [reply apigee_m10n.module]: Added in m10n module

File: src/Monetization.php (hunk 2 above)
Comment (kedarkhaire) [reply src/Monetization.php]: Function called from m10n teams module

File: modules/apigee_m10n_teams/src/MonetizationTeams.php [hunk 3]
```diff
@@ -304,4 +306,33 @@ public function acceptLatestTermsAndConditions(string $company_id): ?LegalEntity
...
+  /**
+   * {@inheritdoc}
+   */
+  public function apiProductTeamAssignmentAccess(ApiProductInterface $api_product, TeamInterface $team, AccountInterface $account): AccessResultInterface {
+    // Cache results for this request.
+    static $eligible_product_cache = [];
+    $company_id = $team->getDisplayName();
+
+    if (!isset($eligible_product_cache[$company_id])) {
+      // kint($this->sdkConnector->getOrganization());
```
Comment (divya-intelli) [inline modules/apigee_m10n_teams/src/MonetizationTeams.php]: Please remove this commented code.

File: modules/apigee_m10n_teams/src/TeamSdkControllerFactory.php [hunk 4]
```diff
@@ -81,4 +82,20 @@ public function teamBalanceController($team_id): CompanyPrepaidBalanceController
...
   }
 
+  /**
+   * {@inheritdoc}
+   */
+  public function companyApiProductController(string $company_id): ApiProductController {
+    if (empty($this->controllers[__FUNCTION__][$company_id])) {
+      // Don't assume the bucket has been initialized.
+      $this->controllers[__FUNCTION__] = $this->controllers[__FUNCTION__] ?? [];
+      // Create a new balance controller.
```
Comment (divya-intelli) [inline modules/apigee_m10n_teams/src/TeamSdkControllerFactory.php]: balance?

File: modules/apigee_m10n_teams/src/TeamSdkControllerFactoryInterface.php [hunk 5]
```diff
@@ -61,4 +62,15 @@ public function companyTermsAndConditionsController(string $company_id): Company
    */
   public function teamBalanceController($team_id): CompanyPrepaidBalanceControllerInterface;
 
+  /**
+   * Creates a team prepaid balance controller.
```
Comment (divya-intelli) [inline modules/apigee_m10n_teams/src/TeamSdkControllerFactoryInterface.php]: balance?

**Review by divya-intelli — APPROVED** [review]
LGTM

**Pull request merged**